SayTheirNamesTests/UtilsTests/PaginatorTests.swift

```diff
@@ -0,0 +1,200 @@
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+import XCTest
+@testable import Say_Their_Names
+
+class PaginatorTests: XCTestCase {
+
+    private enum TestData {
+        static let firstPage = DummyPage(all: [5, 6, 7], link: Link(first: "firstPage_first",
+                                                                    last: "firstPage_last",
+                                                                    prev: "firstPage_prev",
+                                                                    next: "firstPage_next"))
+        static let secondPage = DummyPage(all: [10, 20, 30, 40], link: Link(first: "secondPage_first",
+                                                                            last: "secondPage_last",
+                                                                            prev: "secondPage_prev",
+                                                                            next: "secondPage_next"))
+        static let lastPage = DummyPage(all: [8, 9], link: Link(first: "thirdPage_first",
+                                                                 last: "thirdPage_last",
+                                                                 prev: "thirdPage_prev",
+                                                                 next: nil))
+
+        static let emptyPage = DummyPage(all: [], link: Link(first: "", last: "", prev: nil, next: nil))
+    }
+    
+    private struct DummyPage: PaginatorResponsePage {
+        var all: [Int]
+        var link: Link
+    }
+    
+    func testInit() {
+        let sut = Paginator<Int, DummyPage> { (_, _) in
+            XCTFail("Loader souldn't be called on init")
+        }
+        XCTAssertEqual(sut.state, .notLoading)
+    }
+    
+    func testBasicLoad() {
+        let expectation = self.expectation(description: "Wait for async loading operation")
+
+        let sut = Paginator<Int, DummyPage> { (link, completion) in
+            XCTAssertNil(link)
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+                completion(.success(TestData.firstPage))
+                expectation.fulfill()
+            }
+        }
+        
+        sut.firstPageDataLoadedHandler = { (data) in
+            XCTAssertEqual(data, [5, 6, 7])
+        }
+        sut.subsequentPageDataLoadedHandler = { (data) in
+            XCTFail("Shouldn't be called")
+        }
+        XCTAssertEqual(sut.state, .notLoading)
+        
+        sut.loadNextPage()
+        XCTAssertEqual(sut.state, .loading)
+        self.waitForExpectations(timeout: 1, handler: nil)
+        XCTAssertEqual(sut.state, .notLoading)
+    }
+    
+    func testLoadFailing() {
```

Comment:
Can the test function name be more descriptive?

Reply:
I don't want to push another change as that will dismiss all reviews again, but it's a good point so I'll fix that some other time soon 🙏